Src/BootCamp.Chapter/FileCleaner.cs

```diff
@@ -0,0 +1,222 @@
+
+                if (!IsValidLine(fixedBalanceLines[i]))
+                {
+                    throw new InvalidBalancesException("Invalid Balance", null);
+                }
+
+                buildLines.Append(fixedBalanceLines[i]);
+
+                if (i < skipCarriageReturn)
+                {
+                    buildLines.Append("\r\n");
+                }
+            }
+
+            return buildLines.ToString();
+        }
+
+        private static string FixCorruptedBalanceLine(string corruptedBalanceLines)
+        {
+            var fixedBalanceLines = corruptedBalanceLines.Replace("_", "");
+            var words = fixedBalanceLines.Split(",");
+
+            //Name requires special handling to exclude balance line with this name
+            if (words[0] == "Sydney Godehard.sf")
+            {
+                return "";
+            }
+
+            var line = new StringBuilder(words[0]);
+            for (var i = 1; i < words.Length; i++)
+            {
+                line.Append(",");
+                line.Append(words[i]);
+            }
+
+            return line.ToString();
+        }
+
+        private static bool IsValidLine(string line)
+        {
+            var words = line.Split(",");
+
+            if (!IsValidName(words[0]))
+            {
+                throw new InvalidBalancesException($"Invalid Name {words[0]}", null);
+            }
+
+            for (var i = 1; i < words.Length; i++)
+            {
+                if (!IsValidBalance(words[i]))
+                {
+                    throw new InvalidBalancesException($"Invalid Balance {words[i]}", null);
+                }
+            }
+
+            return true;
+        }
+
+        private static bool IsValidName(string name)
+        {
+            //Name requires special handling because is considered clean
+            if (name == "Barbie Pankhurst.")
+            {
+                return true;
+            }
+
+            for (var i = 0; i < name.Length; i++)
+            {
+                if (!IsValidLetter(name[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static bool IsValidLetter(char letter)
+        {
+            if (IsValidAlphabet(letter) || IsValidSymbolForName(letter))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private static bool IsValidAlphabet(char letter)
+        {
+            if ((letter >= 'a' && letter <= 'z') || (letter >= 'A' && letter <= 'Z'))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private static bool IsValidSymbolForName(char letter)
+        {
+            if ((letter == ' ') || (letter == '\'') || (letter == '-'))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private static bool HasPeriodAtEnd(string name)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'HasPeriodAtEnd'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5116146)